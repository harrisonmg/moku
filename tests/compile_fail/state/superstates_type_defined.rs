use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod state_machine {}

    struct Top;
    impl TopState<BlinkyState> for Top {}

    struct Bottom;

    #[superstate(Top)]
    impl State<BlinkyState> for Bottom {
        type Superstates<'a> = std::marker::PhantomData<&'a ()>;
    }
}

fn main() {}
