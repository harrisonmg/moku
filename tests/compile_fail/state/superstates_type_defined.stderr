error: the `State::Superstates` associated type must not be manually defined
  --> tests/compile_fail/state/superstates_type_defined.rs:17:9
   |
17 |         type Superstates<'a> = std::marker::PhantomData<&'a ()>;
   |         ^^^^
